index.js

```diff
@@ -83,11 +85,27 @@ module.exports = class NoiseSecretStream extends Duplex {
   }
 
   setTimeout (ms) {
+    if (!ms) ms = 0
+
     this._clearTimeout()
-    this._timeoutMs = ms || 0
+    this._timeoutMs = ms
 
     if (!ms || this.rawStream === null) return
-    this._timeout = new Timeout(ms, destroyTimeout, this)
+
+    this._timeout = Timeout.once(ms, destroyTimeout, this)
+    this._timeout.unref()
+  }
+
+  setKeepAlive (ms) {
+    if (!ms) ms = 0
+
+    if (this._timeoutMs === 0) this.setTimeout(2 * ms)
```

Comment:
Is it intuitive to have these coupled? Explicitness might be better i.e
```
stream.setKeepAlive(KEEP_ALIVE)
stream.setTimeout(2 * KEEP_ALIVE)
```

If you call setKeepAlive, you're trying to prevent the remote from timing out, not necessarily specifying a local timeout condition right? By default I'd intuitively expect it to ping every `ms` interval but not time out

Reply:
Yuh fair, I'll change that.

Reply:
Fixed